docs/pages/reference/helm-reference/teleport-kube-agent.mdx

```diff
@@ -17,9 +34,29 @@ The `teleport-kube-agent` chart can run any or all of three Teleport services:
 | [`application_service`](../../application-access/guides.mdx) | `app` | Uses Teleport to handle authentication with and proxy access to web-based applications |
 | [`database_service`](../../database-access/guides.mdx) | `db` | Uses Teleport to handle authentication with and proxy access to databases |
 
-This reference details available values for the `teleport-kube-agent` chart.
-
-(!docs/pages/includes/backup-warning.mdx!)
+### Kubernetes resources
+
+The `teleport-kube-agent` chart deploys the following Kubernetes resources:
+
+|Kind|Default Name|Description|When Deployed|
+|---|---|---|---|
+|`StatefulSet`|The release name|Running a user-configured Teleport pod.|Always.|
+|`Secret`|`secretName` (default: ` teleport-kube-agent-join-token`)|Used for managing the state of the Teleport pod.|`authToken` or `joinParams.tokenName` is provided.|
+|`Deployment`|The release name|Runs a user-configured Teleport pod.|`storage.enabled` is `false` and the chart is being upgraded. Fresh installs will deploy a `StatefulSet` instead.|
+|`Role`|The `roleName` option, if given, or the release name.|Used to manage the state of the Teleport pod via Kubernetes secrets.|Always.|
+|`ClusterRole`|`clusterRoleName`, if given, or the release name.|Allows impersonating users, groups, and service accounts, getting pods, and creating [`SelfSubjectAccessReview`s](https://www.pulumi.com/registry/packages/kubernetes/api-docs/authorization/v1/selfsubjectaccessreview/) so the Teleport pod can manage access to resources in its Kubernetes cluster.|Always.|
+|`ClusterRoleBinding`|`clusterRoleBindingName`, if provided, or the release name|Enables the Teleport pod to manage access to resources in the Kubernetes cluster.|Always.|
+|`RoleBinding`|`roleBindingName`, if given, or the release name|Enables the Teleport pod to manage access to resources in the Kubernetes cluster.|Always.|
+|`ServiceAccount`|`serviceAccountName`, if given, or the release name|Enables the Teleport pod to manage access to resources in the Kubernetes cluster.|`serviceAccount.create` is `true`|
+|`PodDisruptionBudget`|The release name|Ensure high availability for the Teleport pod.|`highAvailability.podDisruptionBudget.enabled` is `true`.|
+|`ServiceAccount`|The release name, suffixed by `-hook`|Used to delete legacy `Deployment`s in order to deploy a `StatefulSet` instead.|If the `teleport-kube-agent` release contains a legacy `Deployment` resource.|
+|`Role`|The release name, suffixed by `-hook`|Used to delete legacy `Deployment`s in order to deploy a `StatefulSet` instead.|If the `teleport-kube-agent` release contains a legacy `Deployment` resource.|
```

Comment:
We should mention that every `*-hook` resources are removed once the upgrade finishes